sql/src/main/java/io/crate/analyze/repositories/RepositorySettingsModule.java

```diff
@@ -50,29 +49,19 @@
     private static final String S3 = "s3";
 
     private static final TypeSettings FS_SETTINGS = new TypeSettings(
-        ImmutableMap.of("location", FsRepository.LOCATION_SETTING),
-        ImmutableMap.of(
-            "compress", FsRepository.COMPRESS_SETTING,
-            "chunk_size", FsRepository.CHUNK_SIZE_SETTING
-        ));
+        convertSettingListToMap(FsRepository.settingsToValidate()),
+        convertSettingListToMap(FsRepository.optionalSettingsToValidate())
+    );
 
     private static final TypeSettings URL_SETTINGS = new TypeSettings(
-        ImmutableMap.of("url", URLRepository.URL_SETTING),
-        Collections.emptyMap());
+        convertSettingListToMap(URLRepository.settingsToValidate()),
+        Map.of()
+    );
 
     private static final TypeSettings HDFS_SETTINGS = new TypeSettings(
-        Collections.emptyMap(),
-        ImmutableMap.<String, Setting>builder()
-            .put("uri", Setting.simpleString("uri", Setting.Property.NodeScope))
-            .put("security.principal", Setting.simpleString("security.principal", Setting.Property.NodeScope))
-            .put("path", Setting.simpleString("path", Setting.Property.NodeScope))
-            .put("load_defaults", Setting.boolSetting("load_defaults", true, Setting.Property.NodeScope))
-            .put("concurrent_streams", Setting.intSetting("concurrent_streams", 5, Setting.Property.NodeScope))
-            .put("compress", Setting.boolSetting("compress", true, Setting.Property.NodeScope))
-            // We cannot use a ByteSize setting as it doesn't support NULL and it must be NULL as default to indicate to
-            // not override the default behaviour.
-            .put("chunk_size", Setting.simpleString("chunk_size"))
-            .build()) {
+        Map.of(),
+        // tmp comment for review : left out 'concurrent_streams' as I didn't see any use of it
+        convertSettingListToMap(HdfsRepository.settingsToValidate())) {
```

Comment:
are these optional? if so we probably have to name the method accordingly to be consistent with the TypeSettings initialized above.